rust/cardano-blockchain-types/src/metadata/cip36/key_registration.rs

```diff
@@ -94,64 +95,72 @@ impl Decode<'_, ()> for Cip36KeyRegistration {
                 match key {
                     Cip36KeyRegistrationKeys::VotingKey => {
                         if found_keys.contains(&key) {
```

Comment:
Suggest moving each `key` processing to its own function will make the code clearer and also eliminate the need for the clippy allow above.